### PR TITLE
fix: transform deprecated and description properties correctly

### LINKED ISSUE #26: Inconsistent http operation parameter description path
`IHttpParam` has description property on the root of the parameter: https://github.com/stoplightio/types/blob/master/src/http-spec.ts#L75

OpenAPI v2 & v3 have the description property on the root of the parameter: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#fixed-fields-10

oas3 `translateParameterObject` is putting the description on the root: https://github.com/stoplightio/http-spec/blob/master/src/oas3/transformers/request.ts#L26

oas2 `buildSchemaForParameter` is incorrectly putting the description inside the `schema` property: https://github.com/stoplightio/http-spec/blob/master/src/oas2/transformers/params.ts#L182

I think a good test overall for this repo would be to run all the functions against the same API defined in both OpenAPI v2 and an OpenAPI v3 and make sure the http services and http operations that are produced are exactly the same.

You can see this in Studio using the tutorial files. Just look at an operation from the oas3 petstore vs an operation from the oas2 todos:

Stoplight_Studio

```json
{
  "id": "?http-operation-id?",
  "iid": "listPets",
  "method": "get",
  "path": "/pets",
  "summary": "List all pets",
  "servers": [
    {
      "url": "http://petstore.swagger.io/v1",
      "variables": {}
    }
  ],
  "request": {
    "body": {
      "contents": []
    },
    "headers": [],
    "query": [
      {
        "name": "limit",
        "description": "How many items to return at one time (max 100)",
        "schema": {
          "type": "integer",
          "format": "int32"
        },
        "examples": []
      }
    ],
    "cookie": [],
    "path": []
  },
  "tags": [
    {
      "name": "pets"
    }
  ],
  "security": []
}
```

Stoplight_Studio

```json
{
  "id": "?http-operation-id?",
  "iid": "PUT_todos",
  "method": "put",
  "path": "/todos/{todoId}",
  "summary": "Update Todo",
  "servers": [
    {
      "url": "http://todos.stoplight.io"
    }
  ],
  "request": {
    "query": [
      {
        "schema": {
          "type": "string",
          "default": "300",
        },

        "name": "limit",
        "style": "form"
      },
      {
        "schema": {
          "type": "boolean",
          "description": "Only return completed"
        },
        "name": "completed",
        "style": "form"
      },
      {
        "schema": {
          "type": "string",
          "enum": [
            "something",
            "another"
          ]
        },
        "name": "type",
        "style": "form"
      }
    ],
    "headers": [
      {
        "schema": {
          "type": "string",
          "description": "Your Stoplight account id"
        },
        "name": "account-id",
        "style": "simple",
        "required": true
      },
      {
        "schema": {
          "type": "string",
          "default": "123"
        },
        "name": "apikey",
        "style": "simple",
        "required": true
      }
    ],
    "path": [
      {
        "schema": {
          "type": "string"
        },
        "name": "todoId",
        "style": "simple",
        "required": true
      }
    ]
  },
  "tags": [
    {
      "name": "Todos"
    }
  ],
  "security": [
    [],
    [
      {
        "type": "apiKey",
        "name": "apikey",
        "in": "query",
        "description": "Use `?apikey=123` to authenticate requests. It's super secure."
      }
    ]
  ]
}
```

### LINKED ISSUE #42: Deprecated property is missing
**Describe the bug**

Given the following OpenAPI:

```yaml
openapi: 3.0.0
info:
  title: Test
  version: '1.0'
servers:
  - url: 'http://localhost:3000'
paths:
  /test:
    get:
      summary: Your GET endpoint
      tags: []
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: objec
                properties:
                  foo:
                    type: string
                    default: bar
                    deprecated: true
                    enum:
                      - foo
                    format: date
                    example: doo
      operationId: get-test
      deprecated: true
      parameters:
        - schema:
            type: string
          in: query
          name: query-param
          deprecated: true
components:
  schemas: {}
```

I expect to see the deprecated property on the HttpOperation root, parameter, and request/response schema properties as described in the [OpenAPI Spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md)

### PR DESCRIPTION
**Pull request by ssspear**
* fixes #42 
* fixes #26

### CODE REVIEW

File: src/oas3/__tests__/operation.test.ts [hunk 1]
```diff
@@ -1,6 +1,72 @@
 import { transformOas3Operation } from '../operation';
 
 describe('transformOas3Operation', () => {
+  test('should return deprecated property', () => {
```
Comment (P0lip) [inline src/oas3/__tests__/operation.test.ts]: Could we narrow down the assertion? Large snapshots are difficult to follow. :laughing:

File: src/oas2/transformers/params.ts [hunk 2]
```diff
@@ -165,6 +165,7 @@ export function translateToQueryParameter(query: QueryParameter): IHttpQueryPara
 export function translateToPathParameter(parameter: PathParameter): IHttpPathParam {
   return (pickBy({
     ...buildSchemaForParameter(parameter),
+    description: get(parameter, 'description'),
```
Comment (P0lip) [inline src/oas2/transformers/params.ts]:
```suggestion
    description: parameter?.description,
```

File: src/oas2/transformers/__tests__/params.test.ts [hunk 3]
```diff
@@ -157,7 +157,6 @@ describe('params.translator', () => {
         schema: {
           properties: {
             arr: {
-              description: 'desc',
```
Comment (P0lip) [inline src/oas2/transformers/__tests__/params.test.ts]: Are we sure we want these to be removed?

File: src/oas2/transformers/__tests__/params.test.ts [hunk 4]
```diff
@@ -235,5 +232,17 @@ describe('params.translator', () => {
...
+    test('should render description in root', () => {
+      expect(
+        translateToPathParameter({
+          required: true,
+          description: 'descr',
+          name: 'name',
+          in: 'path',
+          type: 'string',
+        }),
+      ).toMatchSnapshot();
```
Comment (P0lip) [inline src/oas2/transformers/__tests__/params.test.ts]:
```suggestion
      ).toHaveProperty('description', 'descr');
```

File: src/oas2/transformers/__tests__/params.test.ts (hunk 3 above)
Comment (ssspear) [reply src/oas2/transformers/__tests__/params.test.ts]: @lottamus Thoughts?

Comment (XVincentX) [reply src/oas2/transformers/__tests__/params.test.ts]: I believe these are used around? Prism's not — but I think they should not be removed.

File: src/oas2/transformers/params.ts [hunk 5]
```diff
@@ -165,6 +165,7 @@ export function translateToQueryParameter(query: QueryParameter): IHttpQueryPara
 export function translateToPathParameter(parameter: PathParameter): IHttpPathParam {
   return (pickBy({
     ...buildSchemaForParameter(parameter),
+    description: parameter?.description,
```
Comment (XVincentX) [inline src/oas2/transformers/params.ts]: How is this different than the previous statement down on line 182?

File: src/oas3/__tests__/operation.test.ts (hunk 1 above)
Comment (XVincentX) [inline src/oas3/__tests__/operation.test.ts]: I think this test is checking out a micro functionality from a too high level perspective (entire oas3 operation); I would just remove this.

File: src/oas2/transformers/params.ts (hunk 5 above)
Comment (ssspear) [reply src/oas2/transformers/params.ts]: This resolves the following issue of inconsistent http http operation parameter description path: https://github.com/stoplightio/http-spec/issues/26

**Pull request merged**